### PR TITLE
emtydir removed #206

### PR DESCRIPTION
**Pull request by vktr-brlv**

### CODE REVIEW

File: e2e/e2e_test.go [hunk 1]
```diff
@@ -111,7 +111,7 @@ func TestCreatesInsecureCluster(t *testing.T) {
 	sb := testenv.NewDiffingSandbox(t, env)
 	sb.StartManager(t, controller.InitClusterReconcilerWithLogger(testLog))
 
-	b := testutil.NewBuilder("crdb").WithNodeCount(3).WithEmptyDirDataStore()
+	b := testutil.NewBuilder("crdb").WithNodeCount(3).WithHostPathDataStore("/mnt/data", nil)
```
Comment (vktr-brlv) [inline e2e/e2e_test.go:114]: Hey @chrislovecnm ,
I have question here. This test fails b/c kind-test cluster is running single cluster node, and sharing hostpath, so pod failing to launch with error 
`E210120 13:01:42.934643 1 cli/error.go:233  could not cleanup temporary directories from record file: could not lock temporary directory /cockroach/cockroach-data/cockroach-temp150903260, may still be in use: IO error: While lock file: /cockroach/cockroach-data/cockroach-temp150903260/TEMP_DIR.LOCK: Resource temporarily unavailable`
Should we change volume type for this case to pvc? Or maybe run multiple node test cluster? WDYT?

Comment (chrislovecnm) [reply e2e/e2e_test.go:114]: Already started to work on this PR

Comment (chrislovecnm) [reply e2e/e2e_test.go:114]: https://github.com/cockroachdb/cockroach-operator/pull/266

Comment (chrislovecnm) [reply e2e/e2e_test.go:114]: Yes, we have not tested hostDir.  Does it work on a cluster?

Comment (vktr-brlv) [reply e2e/e2e_test.go:114]: yes, single node works fine, in multi node container fails with error above when running on single k8s instance with error above. Which makes sense.

Comment (chrislovecnm) [reply e2e/e2e_test.go:114]: Can we remove this please?  Let's move this to PVC

**Comment by chrislovecnm** [comment]
@vktr-brlv what is the status on this?

**Comment by chrislovecnm** [comment]
Closing because we did it in #420

**Pull request closed without merging**